test/e2e/node_feature_discovery_test.go

```diff
@@ -169,9 +169,12 @@ var _ = SIGDescribe("NFD master and worker", func() {
 
 		Context("when deploying a single nfd-master pod", Ordered, func() {
 			var (
-				crds      []*apiextensionsv1.CustomResourceDefinition
-				extClient *extclient.Clientset
-				nfdClient *nfdclient.Clientset
+				crds                      []*apiextensionsv1.CustomResourceDefinition
+				extClient                 *extclient.Clientset
+				nfdClient                 *nfdclient.Clientset
+				isDefaultMasterDeployment = true
+				configMapName             = "nfd-master-conf"
+				configMapKey              = "nfd-master.conf"
```

Comment:
I'd suggest to make this more generic e.g. with smth like

```suggestion
				customMasterPodSpecOpts              =  *[]testpod.SpecOption
```

With this the test can customize the nfd-master pod as they like. Use the default opts if `customMasterPodSpecOpts == nil`, otherwise use the spec opts provided by it. WDYT?

Reply:
@marquiz the configmap key and name are needed to create a new configmap, which is unrelated to the master deployment, so we can't put them there. wdyt?

Reply:
> the configmap key and name are needed to create a new configmap, which is unrelated to the master deployment, so we can't put them there. wdyt?

Yup, but you should be able to create the configmap in the BeforeEach() function of the specific test that needs it, and set the PodSpecOpts accordingly, right?

Reply:
Yes, we are already doing that in the `JustBeforeEach`, but I think we still need the configMapName and configMapKey even outside the `JustBeforeEach` clause, that's why we need it in the `var` section.

Reply:
You shouldn't need those. You should be able to configure that with the `SpecWithConfigMap()` podSpecOpt(?)

Reply:
You wouldn't need to have those as "top level" variables. Just hard code them (as you know the names in the test) or specify variables on the inner Context-.